File: src/main/java/com/flexpoker/game/query/handlers/GameCreatedEventHandler.java
package com.flexpoker.game.query.handlers;

import com.flexpoker.framework.event.EventHandler;
import com.flexpoker.framework.pushnotifier.PushNotificationPublisher;
import com.flexpoker.game.command.events.GameCreatedEvent;
import com.flexpoker.game.query.dto.GameInListDTO;
import com.flexpoker.game.query.dto.GameStage;
import com.flexpoker.game.query.repository.GameListRepository;
import com.flexpoker.login.repository.LoginRepository;
import com.flexpoker.pushnotifications.GameListUpdatedPushNotification;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Component;

import javax.inject.Inject;

@Component
public class GameCreatedEventHandler implements EventHandler<GameCreatedEvent> {

    private final LoginRepository loginRepository;

    private final GameListRepository gameListRepository;

    private final PushNotificationPublisher pushNotificationPublisher;

    @Inject
    public GameCreatedEventHandler(LoginRepository loginRepository,
            GameListRepository gameListRepository,
            PushNotificationPublisher pushNotificationPublisher) {
        this.loginRepository = loginRepository;
        this.gameListRepository = gameListRepository;
        this.pushNotificationPublisher = pushNotificationPublisher;
    }

    @Async
    @Override
    public void handle(GameCreatedEvent event) {
        handleGameListRepository(event);
        handlePushNotifications();
    }

    private void handleGameListRepository(GameCreatedEvent event) {
        var createdByUsername = loginRepository.fetchUsernameByAggregateId(event.getCreatedByPlayerId());
        var gameInListDTO = new GameInListDTO(event.getAggregateId(),
                event.getGameName(), GameStage.REGISTERING.toString(), 0,
                event.getNumberOfPlayers(), event.getNumberOfPlayersPerTable(),
                event.getNumberOfMinutesBetweenBlindLevels(),
                event.getNumberOfSecondsForActionOnTimer(),
                createdByUsername, event.getTime().toString());

        gameListRepository.saveNew(gameInListDTO);
    }

    private void handlePushNotifications() {
        pushNotificationPublisher.publish(GameListUpdatedPushNotification.INSTANCE);
    }

}

File: src/main/kotlin/com/flexpoker/game/query/handlers/GameCreatedEventHandler.kt
package com.flexpoker.game.query.handlers

import com.flexpoker.framework.event.EventHandler
import com.flexpoker.framework.pushnotifier.PushNotificationPublisher
import com.flexpoker.game.command.events.GameCreatedEvent
import com.flexpoker.game.query.dto.GameInListDTO
import com.flexpoker.game.query.dto.GameStage
import com.flexpoker.game.query.repository.GameListRepository
import com.flexpoker.login.repository.LoginRepository
import com.flexpoker.pushnotifications.GameListUpdatedPushNotification
import org.springframework.stereotype.Component
import javax.inject.Inject

@Component
class GameCreatedEventHandler @Inject constructor(
    private val loginRepository: LoginRepository,
    private val gameListRepository: GameListRepository,
    private val pushNotificationPublisher: PushNotificationPublisher
) : EventHandler<GameCreatedEvent> {

    override fun handle(event: GameCreatedEvent) {
        handleGameListRepository(event)
        handlePushNotifications()
    }

    private fun handleGameListRepository(event: GameCreatedEvent) {
        val createdByUsername = loginRepository.fetchUsernameByAggregateId(event.createdByPlayerId)
        val gameInListDTO = GameInListDTO(
            event.aggregateId,
            event.gameName, GameStage.REGISTERING.toString(), 0,
            event.numberOfPlayers, event.numberOfPlayersPerTable,
            event.numberOfMinutesBetweenBlindLevels,
            event.numberOfSecondsForActionOnTimer,
            createdByUsername, event.time.toString()
        )
        gameListRepository.saveNew(gameInListDTO)
    }

    private fun handlePushNotifications() {
        pushNotificationPublisher.publish(GameListUpdatedPushNotification)
    }
}